zho/README.md

```diff
@@ -0,0 +1,174 @@
+<details>
+    <summary>點擊以查看「回調地獄」例子</summary>
+    
+```php
+load_data(function($data) {
+    $init = count($data) === 0 ? init_data(...) : fn($then) => $then($data);
+    $init(function($data) {
+        $output = [];
+        foreach($data as $k => $datum) {
+            processData($datum, function($result) use(&$output, $data) {
+                $output[$k] = $result;
+                if(count($output) === count($data)) {
+                    createQueries($output, function($queries) {
+                        $run = function($i) use($queries, &$run) {
+                            runQuery($queries[$i], function() use($i, $queries, $run) {
+                                if($i === count($queries)) {
+                                    $done = false;
+                                    commitBatch(function() use(&$done) {
+                                        if(!$done) {
+                                            $done = true;
+                                            echo "Done!\n";
+                                        }
+                                    });
+                                    onUserClose(function() use(&$done) {
+                                        if(!$done) {
+                                            $done = true;
+                                            echo "User closed!\n";
+                                        }
+                                    });
+                                    onTimeout(function() use(&$done) {
+                                        if(!$done) {
+                                            $done = true;
+                                            echo "Timeout!\n";
+                                        }
+                                    });
+                                } else {
+                                    $run($i + 1);
+                                }
+                            });
+                        };
+                    });
+                }
+            });
+        }
+    });
+});
+```
+    
+</details>
+如果使用 await-generator ，以上代碼就可以被簡化為：
+
+```php
+$data = yield from load_data();
+if(count($data) === 0) $data = yield from init_data();
+$output = yield from Await::all(array_map(fn($datum) => processData($datum), $data));
+$queries = yield from createQueries($output);
+foreach($queries as $query) yield from runQuery($query);
+[$which, ] = yield from Await::race([
+    0 => commitBatch(),
+    1 => onUserClose(),
+    2 => onTimeout(),
+])
+echo match($which) {
+    0 => "Done!\n",
+    1 => "User closed!\n",
+    2 => "Timeout!\n",
+};
+```
+
+## 使用後的代碼可以維持「backward compatibility 回溯相容性」嗎？
+是的，  await-generator 不會對已有的程式接口（API）造成任何限制。
+你可以將所有涉及 await-generator 的代碼封閉在應用程式的內部。
+但你確實應該把 generation function 直接當作程序接口。
+
+await-generator 會在 `Await::f2c` method 開始進行異步代碼流控制，它接受一般的回調語法，這樣就無需再 <!-- TODO: help wanted-->
```

Comment:
`Await::f2c` doesn't accept callback syntax. I am just saying that you can use this wrapper to create callback style API.